File: Javascript/ejercicio21.js
//Algoritmo ContadorPara
let n;
let factorial;
let i;

n = parseInt(prompt("Ingrese un número entero: "));


if (n < 0) {
    console.log("El factorial no está definido para números negativos.");
    alert("El factorial no está definido para números negativos.");
} else {
    factorial = 1;  // 'factorial' es un número entero
    for (let i = 1; i <= n; i++) {
        factorial = factorial * i ; // Multiplicación de enteros
    }//FinPara
    console.log("El factorial de ", n, " es ", factorial);
    alert("El factorial de ", + n + " es " +factorial);
}
//FinAlgoritmoS

File: Phython/practica/ejercicio21.py
#Algoritmo ContadorPara


entrada = input("Ingrese un número entero: ")

try:
    n = int(entrada) 
    print("¡No ingresaste un número válido!")

    if n < 0:
        print("El factorial no está definido para números negativos.")
    
    else:
        factorial = 1;  # 'factorial' es un número entero
        for  i in range(1,n+1):
            factorial *= i # Multiplicación de enteros
    #FinPara
        print(f"El factorial de {n} es {factorial}")
except ValueError:
    print("¡No ingresaste un número válido!")
#FinAlgoritmoS